tests/utils/uid.tests.ts

```diff
@@ -0,0 +1,32 @@
+import {randomFillSync} from 'crypto';
+import {generateUID} from '../../src/utils/uid';
+
+test('Unique identifier generation', () => {
+    // Make sure we are not messing up global context for the tests
+    expect('crypto' in globalThis).toEqual(false);
+
+    // Create a shim for crypto API
+    // TODO: remove any cast once type declarations are updated
+    const shim1 = jest.fn();
+    globalThis.crypto = {
+        getRandomValues: (buffer) => {
+            shim1();
+            return randomFillSync(buffer);
+        },
+    } as any;
+    const uid1 = generateUID();
+    expect(/^[a-f0-9]{32}$/.test(uid1)).toEqual(true);
+    expect(shim1).toHaveBeenCalled();
+
+    const shim2 = jest.fn();
+    globalThis.crypto = {
+        randomUUID: () => {
+            shim2();
+            return 'a19cc926-bf7f-4d5f-bf9c-202bc7a4c7c6';
+        },
+    } as any;
+    expect(generateUID()).toEqual('a19cc926bf7f4d5fbf9c202bc7a4c7c6');
+    expect(shim2).toHaveBeenCalled();
+
+    globalThis.crypto = undefined;
```

Comment:
Not sure if it will break anything, but use `delete globalThis.crypto` to delete the property, otherwise `'crypto' in globalThis` will return `true`.